akka-typed/src/main/scala/akka/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -75,16 +75,20 @@ import akka.util.OptionVal
       a.SupervisorStrategy.Stop
   }
 
-  override def preStart(): Unit =
+  override def preStart(): Unit = {
     behavior = validateAsInitial(undefer(behavior, ctx))
+    if (!isAlive(behavior)) context.stop(self)
```

Comment:
Overall I don’t feel too good about this behavior interpreter being spread around all places and repeated between the different contexts (ActorCell, ActorAdapter, EffectfulActorContext, …).